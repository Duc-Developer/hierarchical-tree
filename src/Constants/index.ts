export enum TREE_ACTION_TRANSFER {
    ID = 'id',
    PATH = 'path',
    NODE = 'node',
}

export enum DEFAULT_ROOT {
    ID = '__root__',
    NAME = '__root__',
    LEVEL = 0,
}
