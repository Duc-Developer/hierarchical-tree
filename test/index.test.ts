import { test, expect, jest } from 'bun:test';
import TreeFactory from '@src/Components/Factory';

import cars from './car.data.json';
import simple from './simple.data.json';
import persons from './person.data.json';
import { DEFAULT_ROOT } from '@src/Constants';

const generateTree = jest.fn(() => {
    const treeCar = new TreeFactory().produce(cars, { name: 'car', id: 'tree-car' });
    const simpleData = new TreeFactory().produce(simple);
    const treePersons = new TreeFactory().produce(persons);
    return { treeCar, simpleData, treePersons };
});
test('convert cars data to tree', () => {
    const { treeCar } = generateTree();
    const treeJSON = treeCar.toJSON();
    expect(treeJSON).toMatchSnapshot();
    expect(treeJSON.id).toEqual('tree-car');
    expect(treeJSON.name).toEqual('car');
    expect(treeJSON.children[0].id).toEqual('12');
    expect(treeJSON.children[0].name).toEqual('Honda');
    expect(treeJSON.children[0].children.length).toEqual(2);
});

test('convert cities data to tree', () => {
    const { simpleData } = generateTree();
    const treeJSON = simpleData.toJSON();
    expect(treeJSON).toMatchSnapshot();
    expect(treeJSON.id).toEqual(DEFAULT_ROOT.ID);
    expect(treeJSON.name).toEqual(DEFAULT_ROOT.NAME);

    const firstChild = treeJSON.children[0];
    expect(firstChild.id).toEqual('1');
    expect(firstChild.name).toEqual('parent1');
    expect(firstChild.children.length).toEqual(3);
    expect(firstChild.children[0].id).toEqual('11');
    expect(firstChild.children[0].name).toEqual('children1-1');
    expect(firstChild.children[0].parentId).toEqual('1');
    expect(firstChild.children[1].id).toEqual('12');
    expect(firstChild.children[1].name).toEqual('children1-2');
    expect(firstChild.children[1].parentId).toEqual('1');
    expect(firstChild.children[2].id).toEqual('13');
    expect(firstChild.children[2].name).toEqual('children1-3');
    expect(firstChild.children[2].parentId).toEqual('1');

    const secondChild = treeJSON.children[1];
    expect(secondChild.id).toEqual('2');
    expect(secondChild.name).toEqual('parent2');
    expect(secondChild.children.length).toEqual(1);
    expect(secondChild.children[0].id).toEqual('21');
    expect(secondChild.children[0].name).toEqual('children2-1');
    expect(secondChild.children[0].parentId).toEqual('2');
});