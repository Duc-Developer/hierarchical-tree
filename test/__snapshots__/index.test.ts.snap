// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`convert cars data to tree 1`] = `
HierarchicalTree {
  "children": [
    HierarchicalTree {
      "children": [
        HierarchicalTree {
          "children": [],
          "id": "6",
          "level": 2,
          "name": "Audi",
          "parent": undefined,
          "parentId": "12",
          "root": undefined,
        },
        HierarchicalTree {
          "children": [],
          "id": "7",
          "level": 2,
          "name": "Tesla",
          "parent": undefined,
          "parentId": "12",
          "root": undefined,
        },
      ],
      "id": "12",
      "level": 1,
      "name": "Honda",
      "parent": undefined,
      "parentId": undefined,
      "root": undefined,
    },
    HierarchicalTree {
      "children": [
        HierarchicalTree {
          "children": [],
          "id": "11",
          "level": 2,
          "name": "Inova",
          "parent": undefined,
          "parentId": "9",
          "root": undefined,
        },
        HierarchicalTree {
          "children": [],
          "id": "13",
          "level": 2,
          "name": "Vin",
          "parent": undefined,
          "parentId": "9",
          "root": undefined,
        },
      ],
      "id": "9",
      "level": 1,
      "name": "Huyndai",
      "parent": undefined,
      "parentId": undefined,
      "root": undefined,
    },
  ],
  "id": "tree-car",
  "level": 0,
  "name": "car",
  "parent": undefined,
  "parentId": undefined,
  "root": [Circular],
}
`;
